### DISCUSSION TITLE
Solax X3 hybrid gen3 modbus/ethernet

### DISCUSSION THREAD

**Discussion by petrs99** [wills106/homeassistant-solax-modbus#303] [Q&A]
Can somebody help me to figure out how this work for Gen3 model (x3-hybrid 8.0-d-e) I would like to use Wills HA solax modbus integration. Initially I thought i will need to use the RS485 generic USB adaptor, but then I read here, that gen3 has Modbus over TCP. But how does it work for this model? When I tried to unscrew the cover of the inverter it looks like below. The COM port (ethernet?) is connected to the first slot (highlighted) on the left. Is it the correct one? I could not find a proper documentation what each slot is.. only the 3 on the right are with names.

So my question is:-) does anyone have this model or know which one of those 5 slots is the modbus over ethernet port?

I have this user manual for my model -
[https://www.solaxpower.com/wp-content/uploads/2021/04/X3-Hybrid-User-Manual.pdf]
![AF49D752-F756-42F0-8056-C139CE6A8D41](https://user-images.githubusercontent.com/123326135/213935413-2ccd8a92-34bb-47d9-bfc5-3f183858c299.jpeg)

Thanks🙏🙏

**Comment by petrs99** [comment]
This photo I got from Will, which has a completely different setup on his Gen3 single phase version.
![09795D17-5D64-451D-983F-F0C853974B12](https://user-images.githubusercontent.com/123326135/213935604-2279fe1e-5410-44a0-8b35-16ae82ae1af2.jpeg)

**Reply by wills106** [reply]
Where does your BMS connect to? The one between your highlighted Yellow port (Furthest on the left) and your DRM port?

Is that an RJ45 port to the right of your CAN2 port?

Hopefully an owner of the X3 Gen3 can point you in the right direction.

**Reply by petrs99** [reply]
![image](https://user-images.githubusercontent.com/123326135/213936081-5574a59a-0023-41cd-8885-5bc18ad0562d.jpeg)
There are 5 RJ45 ports - no port to the right of CAN2 anymore.

BMS is connected to the 2nd slot from the left.

I also attach the picture from the exterior of the invertor. BMS cable goes to the 2nd slot inside.
COM port slot is empty and is only connected inside the invertor to the first slot.

![image](https://user-images.githubusercontent.com/123326135/213936099-1ece224f-db4a-46c2-96ae-74643c468b84.jpeg)

**Reply by wills106** [reply]
I would assume COM is Ethernet then. But I would personally await confirmation.

I'm surprised to only see a single string of PV on a 8kW Inverter. Or are you planning on adding more PV later?

**Reply by petrs99** [reply]
Ah and if you meant the green connector on the right -  I think it is a meter connection via RS485.

**Reply by wills106** [reply]
> green connector on the right - I think it is a meter connection via RS485

Yea that's your Modbus Power Meter. I meant that black plastic to the right of CAN2. Wasn't sure if it sat on top of another RJ45 socket.

**Reply by petrs99** [reply]
> I would assume COM is Ethernet then. But I would personally await confirmation.
> 
> 
> 
> I'm surprised to only see a single string of PV on a 8kW Inverter. Or are you planning on adding more PV later?

Yes, might be so, but I am not sure as it is not desribed in the manual, an some other manuals for Gen3 models has a different places of the connectors (same as yours).

I currently have 6kwp on the roof (no more place there) and planning 3-4kwp more on the second roof soon.

**Reply by wills106** [reply]
@martin-klima / @mickemartinsson would either of you be able to assist @petrs99 with how to connect their Inverter up over Ethernet?

It looks like it is the COM port which is connected to the furthest left port internally, but it would be appreciated if one of you could confirm that's correct.

**Reply by petrs99** [reply]
Hi Will, thank you are very helpful! I tried to search inside the discussions and wrote both gents an e-mail with a request to help me. I will then summarize here the outcome, when I will have some inputs from them.  Thanks again, much appreciated.

**Reply by petrs99** [reply]
Just coming back here to confirm, that for this model - x3-hybrid 8.0-d-e, the COM port really works as Ethernet (Modbus TCP) and the connection inside the invertor goes to the 1st slot from the left side. Thank you will for very helpful information. I was able also to install your HA integration, which seems to work seamlessly 👍 Thanks

**Reply by Posse9132** [reply]
@petrs99 Hi, I have the Solax X3-Hybrid Gen 3 too and I am trying to get the integration to work. Can you tell me how you managed to integrate it? Did you use a RS485 to RJ45 Ethernet Converter Module (Waveshare for Example) or the RS485 generic USB adaptor?

How exactly did you connect it? I would prefer to use the Waveshare RS485 to RJ45 Ethernet Converter Module but I don't know which cable goes where.

I would appreciate it if you or someone else could help out. Thanks! 🙂

**Reply by wills106** [reply]
Just use an ethernet cable in the COM port if your Inverter is GEN3. There is no need to use a Waveshare and there is nothing to connect it to anyway.

Edit:
Empty LAN in this post is the Ethernet port for Modbus over TCP
https://github.com/wills106/homeassistant-solax-modbus/discussions/303#discussioncomment-4751040

**Reply by Posse9132** [reply]
That worked, thanks a lot! I had to set the IP adress to automatic (it was set to manual by default) and thats why I didn't see the IP address in the router.

**Comment by mickemartinsson** [comment]
Yes, of course.

I’ll check when I get home!


Med vänliga hälsningar
/Mikael Martinsson
Frösön
